services/iam/src/routes/general.js

```diff
@@ -54,6 +54,49 @@ router.post('/login', authMiddleware.authenticate, authMiddleware.accountIsEnabl
 
 });
 
+router.post('/register', async (req, res, next) => {
+    const tenantName = req.body.companyName;
+    const { username ,firstname,password,lastname} = req.body;
+    let tenantId = ""
+    try {
+        const doc = await TenantDAO.findOne({ name: tenantName });
+        if(doc) {
+            return next({message: "Tenant already exists",status: 409})
+        } else {
+        const props = {
+            name: tenantName, 
+            status:CONSTANTS.STATUS.ACTIVE
+        }
+        const tenant = await TenantDAO.create({props});
+        tenantId = tenant._id
```

Comment:
The tenantId variable is created in line 60 and set here, but never used. Can be cleaned out.